tests/blast/test_blast_input.h

```diff
@@ -0,0 +1,1166 @@
+        SEQAN_ASSERT_EQ(length(context.fieldsAsStrings),
+                        13u); // not overwritten because not available
+        for (uint8_t i = 0; i < 13; ++i)
+            SEQAN_ASSERT_EQ((uint32_t)context.fields[i],
+                            (uint32_t)customFields[i]);
+    } else
+    {
+        SEQAN_ASSERT_EQ(length(context.fields),
+                        1u);
+        SEQAN_ASSERT_EQ(length(context.fieldsAsStrings),
+                        12u); // defaults
+        SEQAN_ASSERT_EQ((uint32_t)context.fields[0],
+                        (uint32_t)TField::Enum::STD);
+    }
+    SEQAN_ASSERT_EQ(length(context.otherLines),
+                    0u);
+    SEQAN_ASSERT_EQ(length(context.conformancyErrors),
+                    0u);
+
+    while (onMatch(context, BlastTabular()))
+        skipMatch(it, context, BlastTabular());
+
+    readRecordHeader(r, it, context, BlastTabular());
+
+    SEQAN_ASSERT_EQ(r.qId,
+                    "SHAA005TR  Sample 1 Mate SHAA005TF trimmed_to 22 960");
+    SEQAN_ASSERT_EQ(context.dbName,
+                    "");
+    SEQAN_ASSERT_EQ(context.versionString,
+                    "BLASTX 2.2.26+");
+    SEQAN_ASSERT_EQ(length(r.matches),
+                    0u);
+    SEQAN_ASSERT_EQ(length(context.fields),
+                    1u); // not overwritten because not available
+    SEQAN_ASSERT_EQ(length(context.fieldsAsStrings),
+                    12u); // not overwritten because not available
+    SEQAN_ASSERT_EQ(length(context.otherLines),
+                    1u);
+    SEQAN_ASSERT_EQ(length(context.conformancyErrors),
+                    2u);
+
+    SEQAN_ASSERT_EQ(context.otherLines[0],
+                    " Datacase: /tmp/uniprot_sprot.fasta");
+    SEQAN_ASSERT_EQ(context.conformancyErrors[0],
+                    "No or multiple database lines present.");
+
+    // bottom of file
+    readFooter(context, it, BlastTabular());
+    SEQAN_ASSERT_EQ(length(context.conformancyErrors),
+                    0u);
+
+    // test skipRecordHeader
+    ifstream.close();
+    ifstream.open(toCString(inPath));
+    SEQAN_ASSERT(ifstream.is_open());
+
+    it = directionIterator(ifstream, Input());
+
+    // read header of file
+    readHeader(context, it, BlastTabular());
+    SEQAN_ASSERT(onRecord(context, BlastTabular()));
+    skipRecordHeader(it, context, BlastTabular());
+    SEQAN_ASSERT(onRecord(context, BlastTabular()));
+    skipRecordHeader(it, context, BlastTabular());
+    SEQAN_ASSERT(onRecord(context, BlastTabular()));
+    skipRecordHeader(it, context, BlastTabular());
+    SEQAN_ASSERT(onMatch(context, BlastTabular()));
+
+    ifstream.close();
+}
+
+SEQAN_DEFINE_TEST(test_blast_read_header_tabular_with_header)
+{
+    _test_blast_read_tabular_with_header(false);
+}
+
+SEQAN_DEFINE_TEST(test_blast_read_header_customfields_tabular_with_header)
+{
+    _test_blast_read_tabular_with_header(true);
+}
+
+SEQAN_DEFINE_TEST(test_blast_read_header_tabular_with_header_legacy)
```

Comment:
Please remember to:
* Test only public interfaces
* Stick to **small and generic** unit tests